### PR TITLE
Convert login, logout and statistics pages to templates

### PR DESCRIPTION
**Pull request by PeeHaa**
I converted 3 pages to use templates:

- login (https://bugs.php.net/login.php)
- logout (https://bugs.php.net/logout.php)
- statistics (http://bugs.php.local/stats.php)

Next up would be the report-a-bug page, but that needs some more thought as there is a lot going on on that page.

### CODE REVIEW

File: templates/layout.php [hunk 1]
```diff
@@ -22,7 +22,7 @@
             <a href="/search-howto.php">search howto</a>&nbsp;|&nbsp;
             <a href="/stats.php">statistics</a>&nbsp;|&nbsp;
             <a href="/random">random bug</a>&nbsp;|&nbsp;
-            <?php if ($authIsLoggedIn): ?>
+            <?php if (isset($authIsLoggedIn) && $authIsLoggedIn): ?>
```
Comment (sgolemon) [inline templates/layout.php]: Minor nit; We can use PHP7 syntax on bugs.php.net, so how about:

if ($authIsLoggedIn ?? false):

More concise and avoids repeating the varname.

**Comment by sgolemon** [comment]
Overall, I'm not a fan of the template file approach of assuming vars in the global scope.  It makes tracing the lifetime of a variable much more difficult.  -1

**Comment by PeeHaa** [comment]
> Overall, I'm not a fan of the template file approach of assuming vars in the global scope.

Vars are not in the global scope though. They are specifically passed into the template.

**Comment by PeeHaa** [comment]
ping @sgolemon

**Comment by cmb69** [comment]
IMO, this is much cleaner than the old code (and yes, there are no global variables).

**Comment by peterdd** [comment]
Well, PHP's web-bugs requires a lot of fixes and improvements.
But I think there can be a lot of it done in small steps. It requires some motivated PHP people to regularly review, discuss and merge proposed improvements.

Switching to something else may solve a good portion of current 'not fun' things, but may also add new problems like:

  * Can it handle 80k+ tasks at same/better speed without increasing cost and/or complexity?
  * Adds it dependencies that may hurt in the long run?
  * Who does the migration work and how much 'detail loss' is acceptable?

I saw several tries in the open source world where migrations got stuck and still runs the old ugly but at least somehow working solution.

disclosure 1: I am not part of PHP Group, just a dude who uses PHP.
disclosure 2: I am the current maintainer by of https://github.com/Flyspray/flyspray , also a PHP based bug tracker.

**Comment by derickr** [comment]
bugs.php.net is no longer really used, and hence, we're sunsetting it. Due to that, I don't think it makes sense to merge this patch, which also has conflicts.

**Comment by cmb69** [comment]
Oh, right, further improvements to the bug tracker (especially refactorings) won't make sense anymore.

**Pull request closed without merging**